### PR TITLE
task5 completed

### PR DESCRIPTION
**Pull request by R3dSpyder**
Ticket 5 for review

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,8 +8,8 @@ The API is designed to .....
 
 ## NAMING CONVENTIONS
 
-The controllers begin with 'get', 'post' 'delete'.
-The models begin with 'fetch', 'make', 'remove'.
+The controllers begin with 'get', 'post' 'delete', 'patch'
```
Comment (daviidbartlett) [inline README.md:11]: LOVE THIS

File: __tests__/api.test.js [hunk 2]
```diff
@@ -71,7 +71,7 @@ describe("/api/getCategories", () => {
 ////////////////////// API GET REVIEW OBJECT BY ID//////////////////////////////
 
 describe("/api/getReviewObjectById", () => {
-  it("return a sinlge entity in the form of an object", () => {
+  it("return a single entity in the form of an object", () => {
```
Comment (daviidbartlett) [inline __tests__/api.test.js:74]: spelling is the hardest thing!

File: __tests__/api.test.js [hunk 3]
```diff
@@ -131,3 +131,57 @@ describe("/api/getReviewObjectById", () => {
...
+      .then(({ body }) => {
+        expect(body.msg).toBe("ID out of range");
+      });
+  });
+
+  it("returns an object of some type if passed an object", () => {
+    return request(app)
+      .put("/api/reviews/1")
+      .send({ inc_votes: 1 })
+      .expect(202)
```
Comment (daviidbartlett) [inline __tests__/api.test.js]: 202 is accepted which is related to the accept header, 200 would be more appropriate here

File: mvc/controllers/patchVote.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+const putVote = require("../models/putVote.js");
+
+const patchVote = async (req, res, next) => {
+  if (req.body.inc_votes) {
+    const patch = await putVote(req.body, req.params.review_id).catch((err) => {
+      next(err);
```
Comment (daviidbartlett) [inline mvc/controllers/patchVote.js]: you need to use try catch here

File: mvc/models/putVote.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  return db
+    .query(
+      "UPDATE reviews SET votes = votes + $1 WHERE review_id = $2 RETURNING *;",
+      [inc_votes, id]
+    )
+    .then(({ rows }) => {
+      if (!rows[0]) {
+        return Promise.reject({ status: 404, msg: "ID out of range" });
+      }
+      console.log(rows);
```
Comment (daviidbartlett) [inline mvc/models/putVote.js]: delete console.log

**Review by daviidbartlett — APPROVED** [review]
looks ok, got some dodgy confusion between .catch and async await which can cause bugs! Also I would worry about the routing yet, it's an extension task

File: __tests__/api.test.js (hunk 2 above)
Comment (R3dSpyder) [reply __tests__/api.test.js:74]: haha!

File: mvc/controllers/patchVote.js (hunk 4 above)
Comment (R3dSpyder) [reply mvc/controllers/patchVote.js]: wont the .catch on the end do? I read somewhere its the ideal format for async ??

File: __tests__/api.test.js (hunk 3 above)
Comment (R3dSpyder) [reply __tests__/api.test.js]: done

File: mvc/controllers/patchVote.js (hunk 4 above)
Comment (R3dSpyder) [reply mvc/controllers/patchVote.js]: done

File: mvc/models/putVote.js (hunk 5 above)
Comment (R3dSpyder) [reply mvc/models/putVote.js]: done

**Comment by R3dSpyder** [comment]
Merged after review by David

**Pull request merged**